### ISSUE TITLE
Add timedeltas strategy to datetime extra

### ISSUE THREAD

**Issue by Daenyth** [HypothesisWorks/hypothesis-python#333]
It seems appropriate.

**Comment by Zac-HD** [comment]
@malinoff, @NotBobTheBuilder, do either of you want to finish up your pulls?  If not, I can try integrating the implementation from #441 with the tests and docs from #352 so you both get appropriate credit.

**Comment by NotBobTheBuilder** [comment]
Sure - what needs finishing on mine?

**Comment by Zac-HD** [comment]
I've basically outlined it in #520, where I pulled in some tests and extra docs from @malinoff's implementation.  

- `git pull https://github.com/Zac-HD/hypothesis-python.git timedelta-strategies` to get the merged version.  I've inlined the validation logic, to avoid adding new functions to the namespace which resemble incompatible things, and fixed a few linting issues.
- (optional) I squash some commits to have a cleaner history, if you're comfortable with that.
- Remove `test_can_find_zero_delta`, as it's a poor duplicate of `test_simplifies_towards_zero_delta`
- Change the generation strategy so that timedeltas shrink towards 0, preferring positive values.  If they already do, check that the tests are working (see [my travis build](https://travis-ci.org/HypothesisWorks/hypothesis-python/builds/222081194)).  Note in the documentation that timedelta normalisation means this will give some unintuitive results (eg `timedelta(microseconds=-1)` looks pretty scary).
- Ensure that the tests have complete code coverage; this usually just means passing invalid inputs so that all the validation logic gets exercised.  If it's anything more complicated, I'd be happy to help out.
